src/CachePeer.cc

```diff
@@ -114,6 +117,206 @@ CachePeer::connectTimeout() const
+
+#if USE_DELAY_POOLS
+    if (options.no_delay)
+        yaml << " no-delay";
+#endif
+
+    if (login)
+        yaml << " login=" << login;
+
+    if (mcast.ttl > 0)
+        yaml << " ttl=" << mcast.ttl;
+
+    if (connect_timeout_raw > 0)
+        yaml << " connect-timeout=" << connect_timeout_raw;
+
+    if (connect_fail_limit != PEER_TCP_MAGIC_COUNT)
+        yaml << " connect-fail-limit=" << connect_fail_limit;
+
+#if USE_CACHE_DIGESTS
+    if (digest_url)
+        yaml << " digest-url=" << digest_url;
+#endif
+
+    if (options.allow_miss)
+        yaml << " allow-miss";
+
+    if (options.no_tproxy)
+        yaml << " no-tproxy";
+
+    if (max_conn > 0)
+        yaml << " max-conn=" << max_conn;
+
+    if (standby.limit > 0)
+        yaml << " standby=" << standby.limit;
+
+    if (options.originserver)
+        yaml << " originserver";
+
+    if (domain)
+        yaml << " forceddomain=" << domain;
+
+    if (connection_auth == 0)
+        yaml << " connection-auth=off";
+    else if (connection_auth == 1)
+        yaml << " connection-auth=on";
+    else if (connection_auth == 2)
+        yaml << " connection-auth=auto";
+
+    secure.dumpCfg(yaml, "tls-");
+}
+
+void
+CachePeer::reportStatistics (std::ostream& yaml)
+{
+    yaml <<
+       name << ":\n" <<
+       "  type: " << typeString() << '\n' <<
+       "  host: " << host << '/' << http_port << '/' << icp.port << '\n';
+    yaml << "  flags:";
+    dumpOptions(yaml);
+    yaml << '\n';
+
+    std::vector<Ip::Address> addr;
+    std::copy(addresses, addresses+n_addresses, std::back_inserter(addr));
+    yaml << "  addresses: [ " << AsList(addr).prefixedBy("\"").delimitedBy("\", \"").suffixedBy("\"") << " ]\n";
+
+    yaml << "  status: " << (neighborUp(this) ? "Up" : "Down") << '\n' <<
+       "  fetches: " << stats.fetches << '\n' <<
+       "  open connections: " << stats.conn_open << '\n' <<
+       "  average RTT: " << stats.rtt << " msec\n";
+
+    if (stats.last_query > 0) {
+        yaml << "  last query:" <<
+           (squid_curtime - stats.last_query) << " seconds ago\n";
+    }
+
+    if (stats.last_reply > 0) {
+        yaml << " last reply: " <<
+           (squid_curtime - stats.last_reply) << " seconds ago\n";
+    }
+
+    yaml << "  pings sent: "  << stats.pings_sent << '\n' <<
+       "  pings acked: " << stats.pings_acked << " " <<
+       Math::intPercent(stats.pings_acked, stats.pings_sent) << "%\n";
+
+    yaml << "  replies ignored: " << stats.ignored_replies << " " <<
+       Math::intPercent(stats.ignored_replies, stats.pings_acked) << "%\n";
+
+    auto sectionHeader = AtMostOnce("  histogram of pings acked:\n");
+
+#if USE_HTCP
+    if (options.htcp) {
+        yaml << sectionHeader;
+        yaml << "    htcp misses: " << htcp.counts[0] << " " << Math::intPercent(htcp.counts[0], stats.pings_acked) << "%\n"
+             << "    htcp hits: " << htcp.counts[1] << " " << Math::intPercent(htcp.counts[1], stats.pings_acked) << "%\n";
```

Comment:
If you are making this formally YAML. These two values should be separated and the line use set-brackets to bind them as a pair.
```suggestion
        yaml << "    htcp misses: { count: " << htcp.counts[0] << ", ratio: " << Math::intPercent(htcp.counts[0], stats.pings_acked) << "% }\n"
             << "    htcp hits:  { count: " << htcp.counts[1] << ", ratio: " << Math::intPercent(htcp.counts[1], stats.pings_acked) << "% }\n";
```

... same for the equivalent ICP lines below.

Reply:
.. or the whole thing could be considered as a string from yaml's perspective. I'm fine either way, do you have any strong preference?

Reply:
> These two values should be separated and the line use set-brackets to bind them as a pair.

I am not against this change. FWIW, I said nothing about this in my review because it is not clear to me where to draw the line: Many other places in this output use a single YAML value for what is actually a "compound" value. Ideally, from a machine (i.e. non-human) consumer point of view, even "0 msec" should be converted to avoid post-YAML parsing. We _can_ convert all of these values, but that is quite a bit of work and will reduce output readability for human consumers.

FWIW, if we start converting these things, I will request the removal of the corresponding code duplication like the one in the suggested replacement. Again, that is not a showstopper, it can be done, and it will improve this code quality, but keep that additional work in mind when deciding where to draw the line.

Reply:
I would keep with my prior assessment and not convert, favouring humans.
If we really wanted to target machine consumers, we should probably be looking into supporting Prometheus' native format